addenda99_dishonored_return.go

```diff
@@ -0,0 +1,185 @@
+// Licensed to The Moov Authors under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. The Moov Authors licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ach
+
+import (
+	"strings"
+	"unicode/utf8"
+)
+
+type Addenda99Dishonored struct { // TODO(adam): verfiy this in our books
+	// ID is a client defined string used as a reference to this record.
+	ID string `json:"id"`
+	// RecordType defines the type of record in the block. entryAddendaPos 7
+	recordType string
+	// TypeCode Addenda types code '99'
+	TypeCode string `json:"typeCode"`
+
+	DishonoredReturnReasonCode string
+
+	OriginalEntryTraceNumber string
+
+	OriginalReceivingDFIIdentification string
+
+	ReturnTraceNumber string
+
+	ReturnSettlementDate string
+
+	ReturnReasonCode string
+
+	AddendaInformation string
+
+	TraceNumber string
+
+	// validator is composed for data validation
+	validator
+	// converters is composed for ACH to GoLang Converters
+	converters
+
+	validateOpts *ValidateOpts
+}
+
+// NewAddenda99Dishonored returns a new Addenda99Dishonored with default values for none exported fields
+func NewAddenda99Dishonored() *Addenda99Dishonored {
+	Addenda99Dishonored := &Addenda99Dishonored{
+		recordType: "7",
+		TypeCode:   "99",
+	}
+	return Addenda99Dishonored
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) Parse(record string) {
+	if utf8.RuneCountInString(record) != 94 {
+		return
```

Comment:
Thinking about the future, not this PR but it could be nice to have a logger available and log things like this as a warning (I think it would be a warning / error every time).

Here it silently returns without doing anything. As teams process large volumes of files they periodically may see invalid lines in them, and surfacing visibility / warnings in logs could be useful. That said, this is a bigger change that probably would go across other places in the parsing elsewhere too.

Reply:
Yea I agree. There's room to improve this parsing and remove the silent errors.